apps/srp.c

```diff
@@ -151,14 +153,16 @@ static char *srp_create_user(char *user, char **srp_verifier,
                              char **srp_usersalt, char *g, char *N,
                              char *passout, int verbose)
 {
-    char password[1024];
+    char password[1025];
     PW_CB_DATA cb_tmp;
     char *gNid = NULL;
     char *salt = NULL;
+    int len;
     cb_tmp.prompt_info = user;
     cb_tmp.password = passout;
 
-    if (password_callback(password, sizeof(password), 1, &cb_tmp) > 0) {
+    if ((len = password_callback(password, sizeof(password)-1, 1, &cb_tmp)) > 0) {
+        password[len] = 0;
```

Comment:
I fixed the further coding style issues I catched and pushed the commit.
While at it I also caught an issue with the test to disallow -passin / -passout with multiple users which was not working.